js/main.js

```diff
@@ -1,32 +1,60 @@
 /* global data */
 /* exported data */
-var $input = document.querySelector('.imageUrl');
+
 var $img = document.querySelector('img');
+var $entryButton = document.querySelector('.entry-button');
+var $newButton = document.querySelector('#new');
+var $view = document.querySelectorAll('.view');
+var $p = document.querySelector('p');
+var $ul = document.querySelector('ul');
+var $input = document.querySelector('.imageUrl');
+
 $input.addEventListener('input', updateUrl);
 
 function updateUrl(event) {
   $img.setAttribute('src', $input.value);
 }
 var $form = document.querySelector('form');
-$form.addEventListener('submit', logSubmit);
+$form.addEventListener('submit', formSubmit);
 
-function logSubmit(event) {
-  var entry = null;
+function formSubmit(event) {
   event.preventDefault();
-  entry = {
-    title: document.forms[0].elements.title.value,
-    imageUrl: document.forms[0].elements.imageUrl.value,
-    notes: document.forms[0].elements.notes.value,
-    entryId: data.nextEntryId
-  };
-
-  data.nextEntryId++;
-  data.entries.unshift(entry);
-  $ul.prepend(renderEntry(entry));
+  if (data.editing === null) {
+    var entry = {
+      title: $form.elements.title.value,
+      imageUrl: $form.elements.imageUrl.value,
+      notes: $form.elements.notes.value,
+      entryId: data.nextEntryId
+    };
+    data.nextEntryId++;
+    data.entries.unshift(entry);
+    $ul.prepend(renderEntry(entry));
+    switchViews('entries');
+  } else if (data.editing !== null) {
```

Comment:
This can just be an `else`. If the above `if` statement doesn't run, then you can confidently say that `data.editing` is not `null`.